docs/types/dynamic-dispatch.md

```diff
@@ -36,69 +31,74 @@ one to select, the compiler needs to have a notion of _specificity_, which is
 another.
 
-- Always prefer a member function for both `x.f y` and `f y x` notations.
-- Only member functions, current module's functions, and imported functions are
-  considered to be in scope. Local variable `f` could not be used in the `x.f y`
-  syntax.
-- Selecting the matching function:
-  1. Look up the member function. If it exists, select it.
-  2. If not, find all functions with the matching name in the current module and
-     all directly imported modules. These functions are the _candidates_.
-  3. Eliminate any candidate `X` for which there is another candidate `Y` whose
-     `this` argument type is strictly more specific. That is, `Y` this type is a
-     substitution of `X` this type but not vice versa.
-  4. If not all of the remaining candidates have the same this type, the search
-     fails.
-  5. Eliminate any candidate `X` for which there is another candidate `Y` which
-     type signature is strictly more specific. That is, `Y` type signature is a
-     substitution of `X` type signature.
-  6. If exactly one candidate remains, select it. Otherwise, the search fails.
-
-> The actionables for this section are as follows:
+> [!WARNING]
+> Static compiler selects nothing. The right method to invoke is _selected in the runtime_.
 >
-> - THE ABOVE VERSION IS OLD. NEEDS UPDATING.
-> - The definition of specificity for dispatch candidates (including how it
->   interacts with the subsumption relationship on typesets and the ordering of
->   arguments).
+> - Always prefer a member function for both `x.f y` and `f y x` notations.
+> - Only member functions, current module's functions, and imported functions are
+>   considered to be in scope. Local variable `f` could not be used in the `x.f y`
+>   syntax.
+> - Selecting the matching function:
+>   1. Look up the member function. If it exists, select it.
+>   2. If not, find all functions with the matching name in the current module and
+>      all directly imported modules. These functions are the _candidates_.
+>   3. Eliminate any candidate `X` for which there is another candidate `Y` whose
+>      `this` argument type is strictly more specific. That is, `Y` this type is a
+>      substitution of `X` this type but not vice versa.
+>   4. If not all of the remaining candidates have the same this type, the search
+>      fails.
+>   5. Eliminate any candidate `X` for which there is another candidate `Y` which
+>      type signature is strictly more specific. That is, `Y` type signature is a
+>      substitution of `X` type signature.
+>   6. If exactly one candidate remains, select it. Otherwise, the search fails.
+
+The runtime system of Enso identifies the type of a value in `obj.method_name` invocation.
+It checks the _table of virtual methods_ for given type and finds proper
+implementation of `method_name` to invoke. Should there be no method of given
+name in the value's type (or its supertypes like `Any`) to invoke,
+a `No_Such_Method` panic is raised.
```

Comment:
Shall we also mention there is special dispatch logic for values with warnings attached and dataflow errors?

A dataflow error dispatch first checks if it may call a method on `Error` type.
Similarly for values with attached warnings - I think before we pass the dispatch to the underlying value, we check if `Warning` has a method 'overridden' - this is used for `remove_warnings` etc.